Library/ViewModels/PledgeViewModel.swift

```diff
@@ -449,87 +446,115 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
+      .takeWhen(self.paymentAuthorizationDidFinishSignal)
 
-    self.createBackingError = Signal.merge(
-      createApplePayBackingError,
-      createBackingEventError
+    let createOrUpdateApplePayBackingCompleted = Signal.combineLatest(
+      Signal.merge(
+        updateBackingEvent.filter { $0.isTerminating }.ignoreValues(),
+        createBackingEvent.filter { $0.isTerminating }.ignoreValues()
+      ),
+      paymentAuthorizationDidFinish
     )
 
-    let applePayTransactionCompleted = Signal.combineLatest(project, createApplePayBackingEventSuccess)
-      .takeWhen(createApplePayBackingCompleted)
-      .map(first)
+    let createOrUpdateBackingEventValues = Signal.merge(
+      createBackingEvent.values().map { $0 as StripeSCARequiring },
+      updateBackingEvent.values().map { $0 as StripeSCARequiring }
+    )
 
-    let createBackingTransactionSuccess = project.takeWhen(createBackingEventSuccess)
+    let createOrUpdateBackingEventValuesNoSCA = createOrUpdateBackingEventValues
+      .filter(requiresSCA >>> isFalse)
 
-    self.goToThanks = Signal.merge(applePayTransactionCompleted, createBackingTransactionSuccess)
+    let createOrUpdateBackingDidCompleteNoSCA = submitButtonTapped
+      .takeWhen(createOrUpdateBackingEventValuesNoSCA)
+      .filter(isTrue)
+      .ignoreValues()
 
-    // MARK: - Success/Failure Update
+    let createOrUpdateBackingEventValuesRequiresSCA = createOrUpdateBackingEventValues
+      .filter(requiresSCA)
 
-    let updatePaymentAuthorizationDidFinishSignal = willUpdateApplePayBacking
-      .takeWhen(self.paymentAuthorizationDidFinishSignal)
+    self.beginSCAFlowWithClientSecret = createOrUpdateBackingEventValuesRequiresSCA.map { $0.clientSecret }
+      .skipNil()
 
-    let updateApplePayBackingCompleted = Signal.combineLatest(
-      updateBackingEvent
-        .filter { $0.isTerminating }
-        .ignoreValues(),
-      updatePaymentAuthorizationDidFinishSignal
-    )
+    let didCompleteApplePayBacking = createOrUpdateBackingEventValues
+      .takeWhen(createOrUpdateApplePayBackingCompleted)
 
-    let updateApplePayBackingSuccess = updateBackingEvent.values()
-      .takeWhen(updateApplePayBackingCompleted)
+    let creatingContext = context.filter { $0.isCreating }
 
-    let updateBackingDidCompleteApplePay = updateApplePayBackingSuccess
-      .ignoreValues()
+    #warning("To be removed once new CreateBacking is implemented to handle Apple Pay")
+    let deprecatedCreateApplePayBackingCompleted = Signal.combineLatest(
+      deprecatedCreateApplePayBackingEvent.filter { $0.isTerminating }.ignoreValues(),
+      paymentAuthorizationDidFinish
+    )
 
-    let submitButtonTapped = Signal.merge(
-      self.submitButtonTappedSignal.mapConst(true),
-      willUpdateApplePayBacking.mapConst(false)
+    let deprecatedCreateApplePayBackingCompletedError = deprecatedCreateApplePayBackingCompleted
+      .withLatest(from: deprecatedCreateApplePayBackingEvent.errors())
```

Comment:
For curiosity, why are we using `withLatest(from:)` and not `zip` here? Shouldn't `deprecatedCreateApplePayBackingCompletedError` only emit when both `deprecatedCreateApplePayBackingCompleted` and `deprecatedCreateApplePayBackingEvent.errors()` emit?

Reply:
I wanted to avoid `zip` just because it can often cause sneaky problems, but that's possible true for this case.

I'll look into this along with https://github.com/kickstarter/ios-oss/pull/902#discussion_r337591752.

Reply:
I don't think this is going to be necessary - https://github.com/kickstarter/ios-oss/pull/902#discussion_r337832977.

Also we did `withLatest(from:)` here because the errors emit but we have a race condition between the Apple Pay sheet dismissing and the events emitting, which is why wait for `deprecatedCreateApplePayBackingCompleted`, which is a combined signal to emit and then take the most recently emitted errors.